vcpkg.json

```diff
@@ -38,6 +38,16 @@
       "host": false,
       "platform": "linux"
     },
+    {
+      "name": "gtk",
+      "host": true,
+      "platform": "linux"
+    },
+    {
+      "name": "gtk",
+      "host": false,
+      "platform": "linux"
+    },
```

Comment:
Why gtk in vcpkg? @basilgello you could help confirm.

Reply:
I looked through the PR and the approach is valid. However, bringing full GTK4 for just that purpose is MASSIVE waste of build resources.
I am looking if there is a some other standalone library for monitor things

Reply:
For reference, here it is all that needed for GdkMonitor:

https://chromium.googlesource.com/external/github.com/GNOME/gtk/+/refs/tags/upstream/4.15.4/gdk/gdkmonitor.h
https://chromium.googlesource.com/external/github.com/GNOME/gtk/+/refs/tags/upstream/4.15.4/gdk/gdkmonitor.c
https://chromium.googlesource.com/external/github.com/GNOME/gtk/+/refs/tags/upstream/4.15.4/gdk/gdkmonitorprivate.h

https://chromium.googlesource.com/external/github.com/GNOME/gtk/+/refs/tags/upstream/4.15.4/gdk/x11/gdkmonitor-x11.c
https://chromium.googlesource.com/external/github.com/GNOME/gtk/+/refs/tags/upstream/4.15.4/gdk/x11/gdkmonitor-x11.h

https://chromium.googlesource.com/external/github.com/GNOME/gtk/+/refs/tags/upstream/4.15.4/gdk/broadway/gdkmonitor-broadway.c
https://chromium.googlesource.com/external/github.com/GNOME/gtk/+/refs/tags/upstream/4.15.4/gdk/broadway/gdkmonitor-broadway.h

https://chromium.googlesource.com/external/github.com/GNOME/gtk/+/refs/tags/upstream/4.15.4/gdk/macos/gdkmacosmonitor-private.h
https://chromium.googlesource.com/external/github.com/GNOME/gtk/+/refs/tags/upstream/4.15.4/gdk/macos/gdkmacosmonitor.h
https://chromium.googlesource.com/external/github.com/GNOME/gtk/+/refs/tags/upstream/4.15.4/gdk/macos/gdkmacosmonitor.c

https://chromium.googlesource.com/external/github.com/GNOME/gtk/+/refs/tags/upstream/4.15.4/gdk/wayland/gdkmonitor-wayland.c
https://chromium.googlesource.com/external/github.com/GNOME/gtk/+/refs/tags/upstream/4.15.4/gdk/wayland/gdkmonitor-wayland.h

https://chromium.googlesource.com/external/github.com/GNOME/gtk/+/refs/tags/upstream/4.15.4/gdk/win32/gdkmonitor-win32.c
https://chromium.googlesource.com/external/github.com/GNOME/gtk/+/refs/tags/upstream/4.15.4/gdk/win32/gdkmonitor-win32.h

That's for all supported platforms :)

Reply:
Yes, it is such a small function, but introduce big complexity.

Reply:
I'd isolate the minimal init code, make 1 c or cpp file and build it via vcpkg like oboe-wrapper.

Reply:
> I'd isolate the minimal init code, make 1 c or cpp file and build it via vcpkg like oboe-wrapper.

I can't get your point. Do you mean extract source code of GtkMonitor from gtk? But it's almost coupled with other components of gtk, I don't it's possible.

Reply:
It is possible but I noticed that linux builds use "system" gtk headers. Even Flatpak and AppImage ship libsciter-gtk.so and gtk3.so. Can we use that lib instead of shipping one more instance of gtk4?

Reply:
This fractional scale is support only for gtk >= 4.14

Reply:
> Can we use that lib instead of shipping one more instance of gtk4?

No

Reply:
> I'd isolate the minimal init code, make 1 c or cpp file and build it via vcpkg like oboe-wrapper.

You don't have to use vcpkg. We use vcpkg is for its cross-platform advantage. This is for Linux only. You do not need to use vcpkg.

Reply:
That's why somebody add this.

https://github.com/rustdesk/rustdesk/blob/ae16b8975bdb66c7e74c600797fd98826f107eb7/libs/scrap/build.rs#L8

Reply:
> > I'd isolate the minimal init code, make 1 c or cpp file and build it via vcpkg like oboe-wrapper.
> 
> You don't have to use vcpkg. We use vcpkg is for its cross-platform advantage. This is for Linux only. You do not need to waste time on vcpkg.

I use vcpkg because older ubuntu in CI doesn't support the newer version of gtk.